lab_1_keywords_tfidf/main.py

```diff
@@ -66,7 +87,24 @@ def get_top_n(frequencies: dict[str, Union[int, float]], top: int) -> Optional[l
 
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not (isinstance(frequencies, dict) and isinstance(top, int)
+            and top > 0 and frequencies and isinstance(top, bool) is False):
+        return None
+    items = frequencies.items()
+    for key, value in items:
+        if not isinstance(key, str):
+            return None
+        if not isinstance(value, (float, int)) :
+            return None
+        sorted_items = dict(sorted(items, key=lambda item: item[1], reverse=True))
```

Comment:
having this level of identation, you will sort and return value (and consequently escape the function) after checking validity of just one pair of items

Reply:
done